app/services/doc_auth/error_generator.rb

```diff
@@ -312,29 +328,39 @@ def generate_doc_auth_errors(response_info)
       metrics_error = metrics_error_handler.handle(response_info)
       return metrics_error.to_h if metrics_error.present? && !metrics_error.empty?
 
-      # check selfie error
-      selfie_error_handler = SelfieErrorHandler.new
-      selfie_error = selfie_error_handler.handle(response_info)
+      doc_auth_error_count = doc_auth_error_count(response_info)
+      known_error_count = doc_auth_error_count - unknown_fail_count
+      doc_auth_error_handler = DocAuthErrorHandler.new
+      doc_auth_error = doc_auth_error_handler.handle(response_info, known_error_count)
 
-      # if selfie itself is ok, but we have selfie related error
-      if selfie_error_handler.is_generic_selfie_error?(selfie_error)
-        return selfie_error_handler.selfie_general_failure_error
+      if doc_auth_error.present? && !doc_auth_error.empty?
+        return doc_auth_error.to_h
       end
 
-      # other vendor response detail error
-      liveness_enabled = response_info[:liveness_enabled]
-      alert_error_count = response_info[:doc_auth_result] == 'Passed' ?
-                            0 : response_info[:alert_failure_count]
+      # check selfie error
+      if doc_auth_error_count < 1 && doc_auth_passed_or_attn_with_barcode(response_info)
```

Comment:
I think the count will be `0` if `doc_auth_passed_or_attn_with_barcode` is true?

Reply:
You would think so! [See my comment to Charley here](https://github.com/18F/identity-idp/pull/10450#discussion_r1569208722), as there is one edge case here that it covers.

Reply:
May be following is eaier to understand?

```ruby
failed_count = response_info[:processed_alerts][:failed].count
if failed_count < 1 || doc_auth_passed_or_attn_with_barcode(response_info)
  ####
end  

```

Reply:
Thanks for pushing on this again. It made me re-consider this logic and investigate further.

I found the issue was actually with the yml sample file in the test - we had an "attention" result but no message. I don't think that's a legitimate state. Once I gave it a non-barcode error, I found I was able to simplify this even further as originally pointed out by @charleyf (Sorry I didn't catch this earlier when you brought it up, Charley!).

Fixed in 368e3eeec725ad8a68d9e3f0bd1b35c8dbb1c81e.